tests/hardware/gpio/basic_test_2.cpp

```diff
@@ -0,0 +1,39 @@
+#include <stdint.h>
+#include <util/delay.h>
+#include "../../../mcal/inc/atmega32.h"
+#include "../../../utils/inc/register.h"
+#include "../../../mcal/inc/gpio.h"
+
```

Comment:
It will be great if we agreed to add a comment that describe test. I mean something like arduino example comments.
e.g.
```
/*
 * Hardware Circuit:
 *        Connect Positive pin of LED and negative pin with ground.  
 *        Connect one of push-button pint to A0 and other one to the ground
 * Expected Output:
 *        If the button is pressed then then the led should be turned on.
 *        If the button is released then the led will be turned off
*/
```

Reply:
also it will be good if you care a little bit about code style and aligning :
```
if ( val == kLow)
        {
        gpioA.Write<Pin::kPin0, DigitalLevel::kHigh>();
        }
```
be something like:
```
if (val == kLow) {
    gpioA.Write<Pin::kPin0, DigitalLevel::kHigh>();
}
```

Reply:
Please note PR title will be the title of all commits after merge as we merge with squash so we should care too much about it to make another contributors know what is the main purpose/aim of this PR? 

`Testing Gpio in real hardware (@Mariamahmed44 , @abadr99 )` 

**Note**: 
1) Mentioning name of PR author and reviewer will helps in the future
2) Labels will be useful for search :)

Reply:
@abadr99 Done